stories/module-optimize-settings.stories.js

```diff
@@ -0,0 +1,143 @@
+	...global._googlesitekitLegacyData.modules.optimize,
+	active: true,
+	setupComplete: true,
+};
+
+function Settings( props ) {
+	const {
+		callback,
+		module = global._googlesitekitLegacyData.modules.optimize,
+		isEditing = false,
+		isOpen = true,
+		isSaving = false,
+		error = false,
+		// eslint-disable-next-line no-console
+		handleAccordion = ( ...args ) => console.log( 'handleAccordion', ...args ),
+		// eslint-disable-next-line no-console
+		handleDialog = ( ...args ) => console.log( 'handleDialog', ...args ),
+		// eslint-disable-next-line no-console
+		updateModulesList = ( ...args ) => console.log( 'updateModulesList', ...args ),
+		// eslint-disable-next-line no-console
+		handleButtonAction = ( ...args ) => console.log( 'handleButtonAction', ...args ),
+	} = props;
+
+	return (
+		<WithTestRegistry callback={ callback }>
+			<div style={ { background: 'white' } }>
+				<SettingsModule
+					key={ module.slug + '-module' }
+					slug={ module.slug }
+					name={ module.name }
+					description={ module.description }
+					homepage={ module.homepage }
+					learnmore={ module.learnMore }
+					active={ module.active }
+					setupComplete={ module.setupComplete }
+					hasSettings={ true }
+					autoActivate={ module.autoActivate }
+					updateModulesList={ updateModulesList }
+					handleEdit={ handleButtonAction }
+					handleConfirm
+					isEditing={ isEditing ? { 'optimize-module': true } : {} }
+					isOpen={ isOpen }
+					handleAccordion={ handleAccordion }
+					handleDialog={ handleDialog }
+					provides={ module.provides }
+					isSaving={ isSaving }
+					screenID={ module.screenID }
+					error={ error }
+				/>
+			</div>
+		</WithTestRegistry>
+	);
+}
+
+storiesOf( 'Optimize Module/Settings', module )
+	.add( 'View, closed', () => {
+		filterOptimizeSettings();
+
+		const setupRegistry = ( { dispatch } ) => {
+			dispatch( STORE_NAME ).receiveGetSettings( {} );
+		};
+
+		return <Settings isOpen={ false } module={ completeModuleData } callback={ setupRegistry } />;
+	} )
+	.add( 'View, open with all settings', () => {
+		filterOptimizeSettings();
+
+		const setupRegistry = ( { dispatch } ) => {
+			dispatch( STORE_NAME ).receiveGetSettings( {
+				optimizeID: 'OPT-1234567',
+			} );
+		};
+
+		return <Settings module={ completeModuleData } callback={ setupRegistry } />;
+	} )
+	.add( 'Edit, open with all settings', () => {
+		filterOptimizeSettings();
+
+		const setupRegistry = ( { dispatch } ) => {
+			dispatch( STORE_NAME ).setSettings( {} );
+			dispatch( STORE_NAME ).setOptimizeID( 'OPT-1234567' );
+		};
+
+		return <Settings isEditing={ true } module={ completeModuleData } callback={ setupRegistry } />;
+	} )
+	.add( 'Edit, open with no optimize ID', () => {
+		filterOptimizeSettings();
+
+		const setupRegistry = ( { dispatch } ) => {
+			dispatch( STORE_NAME ).setSettings( {} );
+		};
+
+		return <Settings isEditing={ true } module={ completeModuleData } callback={ setupRegistry } />;
+	} )
```

Comment:
Can we also add a story here that displays the AMP experiment field?

Reply:
:+1: